### PR TITLE
Added Reverse a String in Dart Article

### LINKED ISSUE #460: Add Article for Reverse a String in Dart
See #459

### PR DESCRIPTION
**Pull request by slashdoom**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][1]! For simplicity, please make sure that your pull request includes one and only one contribution.

## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [x] I fixed #460

### Code Snippets

- [ ] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])

### Documentation

- [x] I named the pull request using `Added/Updated <Sample Program> in <Language> Article` format
- [ ] I followed the applicable article template
  - [ ] I added/updated a [language article][4] (i.e. The Python Programming Language)
  - [x] I added/updated a [code snippet article][3] (i.e. Hello World in Perl)
  - [ ] I added/updated a [project article][5] (i.e. Fizz Buzz in Every Language)

### Testing

- [ ] I named the pull request using `Added/Updated <Language>/<Project> Testing` format
- [ ] I followed the testinfo template, if applicable

### CODE REVIEW

File: docs/projects/reverse-a-string/_posts/2019-10-16-dart.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    return input.split('').reversed.join();
+}
+```
+[^1]
+
+---
+
+#### References
+
+[^1]: P. Ryon (slashdoom), “Reverse a String in Dart,” The Renegade Coder, 16-Oct-2019. [Online]. Available: <https://therenegadecoder.com/code/reverse-a-string-in-dart/>. [Accessed: 16-Oct-2019].
```
Comment (jrg94) [inline docs/projects/reverse-a-string/_posts/2019-10-16-dart.md]: It looks like you're referencing an article that doesn't exist. Does that mean you'd like this article cross-posted on my personal blog? I'd be more than happy to do that!

File: archive/d/dart/README.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ Welcome to Sample Programs in Dart!
 ## Sample Programs
 
 - [Hello World in Dart](https://therenegadecoder.com/blog/hello-world-in-dart/)
-- [Reverse String in Dart](https://github.com/TheRenegadeCoder/sample-programs/issues/460)
+- [Reverse String in Dart](https://therenegadecoder.com/code/reverse-a-string-in-dart/)
```
Comment (jrg94) [inline archive/d/dart/README.md]: See comment in the article!

File: docs/projects/reverse-a-string/_posts/2019-10-16-dart.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
+---
+title: Reverse a String in Dart
+layout: default
+last-modified: 2019-10-16
+featured-image:
+tags: [dart, reverse-a-string, string-reversal]
+authors:
+  - P. Ryon (slashdoom)
```
Comment (jrg94) [inline docs/projects/reverse-a-string/_posts/2019-10-16-dart.md]: Side note! Make sure you add yourself to the authors file. You'll need to come up with a username which you'll put here (slashdoom is fine).

https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/_data/authors.yml

File: docs/projects/reverse-a-string/_posts/2019-10-16-dart.md (hunk 1 above)
Comment (slashdoom) [reply docs/projects/reverse-a-string/_posts/2019-10-16-dart.md]: To be honest I just saw the links on so many of the other docs I though it must have been automatically cross-posted and went with it.  I'm happy to have this there as well though.  Thanks for that!

Comment (jrg94) [reply docs/projects/reverse-a-string/_posts/2019-10-16-dart.md]: Yeah, that's fine by me! Right now, it wouldn't get posted until November 1st. So, that link will be dead for about two weeks.

**Pull request merged**